### PR TITLE
protonmail-bridge: 2.3.0 -> 3.0.18

### PR DESCRIPTION
**Pull request by MrFreezeex**
###### Description of changes
Update protonmail bridge to latest version

###### Things done
- Refactor package to use standard nixpkgs go buildmethods
- Update to v3.0
- Enable proxyVendor (needed in newest version)

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by j4m3s-s** [comment]
I've helped with the package update, so can attest it works.

**Comment by j4m3s-s** [comment]
Result of `nixpkgs-review pr 218025` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>protonmail-bridge</li>
  </ul>
</details>

**Review by j4m3s-s — APPROVED** [review]

File: pkgs/applications/networking/protonmail-bridge/default.nix [hunk 1]
```diff
@@ -2,44 +2,50 @@
...
-  version = "2.3.0";
+  version = "3.0.18";
 
   src = fetchFromGitHub {
     owner = "ProtonMail";
     repo = "proton-bridge";
-    rev = "br-${version}";
-    sha256 = "sha256-7p+Q6/BphE/dxNQe+gfcIty6TAWHUcPpvSJWfmf4OQg=";
+    rev = "v${version}";
+    sha256 = "sha256-0gQnMhjwW2NEJwafqndStQ33dIu82lW6ntXFRCpbmm4=";
```
Comment (dotlambda) [inline pkgs/applications/networking/protonmail-bridge/default.nix:11]:
```suggestion
    hash = "sha256-0gQnMhjwW2NEJwafqndStQ33dIu82lW6ntXFRCpbmm4=";
```

File: pkgs/applications/networking/protonmail-bridge/default.nix [hunk 2]
```diff
@@ -2,44 +2,50 @@
...
   };
 
-  vendorSha256 = "sha256-dhrn6xQ0IJzBYeO6ko2PUCO+idopC2An0ylqCnx5jKg=";
+  vendorSha256 = "sha256-lHMcVcaoBwjE2ikEZPeZexC5XvhkAtvHnDci7UAa4vg=";
```
Comment (dotlambda) [inline pkgs/applications/networking/protonmail-bridge/default.nix:14]:
```suggestion
  vendorHash = "sha256-lHMcVcaoBwjE2ikEZPeZexC5XvhkAtvHnDci7UAa4vg=";
```

File: pkgs/applications/networking/protonmail-bridge/default.nix [hunk 3]
```diff
@@ -2,44 +2,50 @@
...
+    "cmd/Desktop-Bridge"
+  ];
+
+  postInstall = ''
+    mv $out/bin/Desktop-Bridge $out/bin/bridge # The cli is named like that in the upstream repo
   '';
 
   meta = with lib; {
     homepage = "https://github.com/ProtonMail/proton-bridge";
     changelog = "https://github.com/ProtonMail/proton-bridge/blob/master/Changelog.md";
```
Comment (dotlambda) [inline pkgs/applications/networking/protonmail-bridge/default.nix:45]:
```suggestion
    changelog = "https://github.com/ProtonMail/proton-bridge/blob/${srv.rev}/Changelog.md";
```

File: pkgs/applications/networking/protonmail-bridge/default.nix [hunk 4]
```diff
@@ -2,44 +2,50 @@
...
-    sha256 = "sha256-7p+Q6/BphE/dxNQe+gfcIty6TAWHUcPpvSJWfmf4OQg=";
+    rev = "v${version}";
+    sha256 = "sha256-0gQnMhjwW2NEJwafqndStQ33dIu82lW6ntXFRCpbmm4=";
   };
 
-  vendorSha256 = "sha256-dhrn6xQ0IJzBYeO6ko2PUCO+idopC2An0ylqCnx5jKg=";
+  vendorSha256 = "sha256-lHMcVcaoBwjE2ikEZPeZexC5XvhkAtvHnDci7UAa4vg=";
 
   nativeBuildInputs = [ pkg-config ];
-
```
Comment (dotlambda) [inline pkgs/applications/networking/protonmail-bridge/default.nix:17]: Why remove the blank line?

Comment (MrFreezeex) [reply pkgs/applications/networking/protonmail-bridge/default.nix:17]: Don't know looks nicer to me, but I reverted back to the way it was before :man_shrugging: .

File: pkgs/applications/networking/protonmail-bridge/default.nix [hunk 5]
```diff
@@ -2,44 +2,51 @@
...
   '';
 
   meta = with lib; {
     homepage = "https://github.com/ProtonMail/proton-bridge";
-    changelog = "https://github.com/ProtonMail/proton-bridge/blob/master/Changelog.md";
+    changelog = "https://github.com/ProtonMail/proton-bridge/blob/${src.rev}/Changelog.md";
     downloadPage = "https://github.com/ProtonMail/proton-bridge/releases";
     license = licenses.gpl3Plus;
-    maintainers = with maintainers; [ lightdiscord ];
+    maintainers = with maintainers; [ lightdiscord mrfreezeex ];
```
Comment (lightdiscord) [inline pkgs/applications/networking/protonmail-bridge/default.nix:49]: Is it possible to remove me from the `maintainers` list ?

Comment (MrFreezeex) [reply pkgs/applications/networking/protonmail-bridge/default.nix]: sure

**Review by drupol — COMMENTED** [review]
LGTM

**Comment by binarin** [comment]
`bridge` executable is also a part of `iproute2` package, which (I think) is always installed on any modern linux system. So probably keeping it under `protonmail-bridge` name would've been better.

**Comment by drupol** [comment]
Nice catch! I agree.

**Comment by MrFreezeex** [comment]
Ah yes indeed and I also just noticed that they indeed it named it bridge in their Makefile but for instance in the deb package it's renamed protonmail-bridge afterwards :man_facepalming: .

**Comment by MrFreezeex** [comment]
Fixed it here: https://github.com/NixOS/nixpkgs/pull/221297 (bundled with a small update as well)

**Pull request merged**